code/modules/food_and_drinks/food.dm

```diff
@@ -47,4 +49,14 @@
 				SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "breakfast", /datum/mood_event/breakfast)
 			last_check_time = world.time
 
+/obj/item/reagent_containers/food/proc/on_brew(datum/source, list/brewing_results)
+	var/list/my_reagents = list()
+	for(var/R in reagents.reagent_list)
+		var/datum/reagent/RE = R
+		my_reagents += RE.type
+	brewing_results["reagents"] = my_reagents
+	brewing_results["booze_power"] = wine_power
+	brewing_results["prefix"] = name
```

Comment:
If you want to return multiple values like this you're better off modifying the barrel from the signal receivers. This sort of thing makes for error prone code. Make some vars on the barrel to keep track of booze power and prefix then just modify them here

Reply:
what? No, go read how the barrel works

Reply:
ditto, this should be in a component and not on food itself, seeing as we're removing /food soon

Reply:
See ninja's explanation for why components aren't optimal earlier, though I'm with you on the idea that it'd be nice to have some external handler for these functions instead of just signal registering.

Reply:
then elements should be used. It's stupid to have this shit on the /food subtype when we're trying to kill it and it just means I have to clean it up in the future.